lib/db/index.js

```diff
@@ -220,7 +225,24 @@ Db.prototype.exec = function (query, options) {
 
   this.server.logger.debug('executing query against db ' + this.name + ': ' + query);
 
-  return this.send('command', data);
+  this.emit("beginQuery", {
+    query: data.query,
+    mode: data.mode,
+    fetchPlan: data.fetchPlan,
+    limit: data.limit,
+    params: data.params
+  });
+
+  var s = Date.now();
+  return this.send('command', data).then(function(res) {
+    var e = Date.now();
```

Comment:
Hmmm good point maybe we should also pass the query object and success/fail to `endQuery`, thoughts?